hamilton/function_modifiers/recursive.py

```diff
@@ -600,3 +603,292 @@ def required_config(self) -> Optional[List[str]]:
+            to pass columns_to_pass in, and we will extract the columns out for you.
+        :param namespace: The namespace of the nodes, so they don't clash with the global namespace
+            and so this can be reused. If its left out, there will be no namespace (in which case you'll want
+            to be careful about repeating it/reusing the nodes in other parts of the DAG.)
+        :param config_required: the list of config keys that are required to resolve any functions. Pass in None\
+            if you want the functions/modules to have access to all possible config.
+        """
+
+        self.subdag_functions = subdag.collect_functions(load_from)
+
+        if select is None:
+            raise ValueError("Please specify at least one column to append or update.")
+        elif isinstance(select, str):
+            self.select = [select]
+        else:
+            self.select = select
+
+        if (pass_dataframe_as is not None and columns_to_pass is not None) or (
+            pass_dataframe_as is None and columns_to_pass is None
+        ):
+            raise ValueError(
+                "You must specify only one of columns_to_pass and "
+                "pass_dataframe_as. "
+                "This is because specifying pass_dataframe_as injects into "
+                "the set of columns, allowing you to perform your own extraction"
+                "from the dataframe. We then execute all columns in the sbudag"
+                "in order, passing in that initial dataframe. If you want"
+                "to reference columns in your code, you'll have to specify "
+                "the set of initial columns, and allow the subdag decorator "
+                "to inject the dataframe through. The initial columns tell "
+                "us which parameters to take from that dataframe, so we can"
+                "feed the right data into the right columns."
+            )
+
+        # TODO: Setting this up to pass in "*" to select all (same as in Polars)
+        if isinstance(columns_to_pass, str):
+            self.initial_schema = [columns_to_pass]
+        else:
+            self.initial_schema = columns_to_pass
+
+        self.dataframe_subdag_param = pass_dataframe_as
+        self.namespace = namespace
+        # This never gets used within the class, but pyspark had it so keeping it here in case we
+        # need to access it from somewhere outside
+        self.upstream_dependency = pd.DataFrame
+
+    # TODO: This is duplicate from extract columns, probably can be combined
+    # TODO: Add async functionality
+    def _create_column_nodes(
+        self, inject_parameter: str, params: Dict[str, Type[Type]]
+    ) -> List[node.Node]:
+        """Helper function to extract columns for the user."""
+        output_type = params[inject_parameter]
+        series_type = registry.get_column_type_from_df_type(output_type)
+
+        out = []
+        for column in self.initial_schema:
+
+            def extractor_fn(
```

Comment:
Ooh we should be able to share logic with `extract_columns`... Might be good to leave this as a TODO note